src/app/modules/submission/encrypt-submission/encrypt-submission.utils.ts

```diff
@@ -359,3 +371,92 @@ export const getPaymentIntentDescription = (
     }
   }
 }
+
+/**
+ * Determines the prefix for a question based on whether it is verified
+ * by MyInfo.
+ * @param response
+ * @param hashedFields Field ids of hashed fields.
+ * @returns the prefix
+ */
+const getMyInfoPrefix = (
+  response: ProcessedFieldResponse,
+  hashedFields: Set<MyInfoKey>,
+): string => {
+  return !!response.myInfo?.attr && hashedFields.has(response._id)
+    ? MYINFO_PREFIX
+    : ''
+}
+
+export const getAnswersForChild = (
+  response: ProcessedChildrenResponse,
+): ProcessedSingleAnswerResponse[] => {
+  const subFields = response.childSubFieldsArray
+  const qnChildIdx = response.childIdx ?? 0
+  if (!subFields) {
+    return []
+  }
+  return response.answerArray.flatMap((arr, childIdx) => {
+    // First array element is always child name
+    const childName = arr[0]
+    return arr.map((answer, idx) => {
+      const subfield = subFields[idx]
+      return {
+        _id: getMyInfoChildHashKey(
+          response._id,
+          subFields[idx],
+          childIdx,
+          childName,
+        ),
+        // qnChildIdx represents the index of the MyInfo field
+        // childIdx represents the index of the child in this MyInfo field
+        // as there might be >1 child for each MyInfo child field if "Add another child" is used
+        question: `Child ${qnChildIdx + childIdx + 1} ${
+          MYINFO_ATTRIBUTE_MAP[subfield].description
+        }`,
+        answer,
+        fieldType: MYINFO_ATTRIBUTE_MAP[subfield].fieldType,
+        isVisible: response.isVisible,
+        myInfo: {
+          attr: subFields[idx] as unknown as MyInfoAttribute,
+        },
+      }
+    })
+  })
+}
+
+export class SubmissionStorageObj {
```

Comment:
might make sense to be a function instead of a class with 1 single method

Reply:
created a new function `formatMyInfoStorageResponseData` in [cf430b0](https://github.com/opengovsg/FormSG/pull/6870/commits/cf430b06b9cf302b5665c1690b91a7fd89c67d87)